composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/presentation/ViewModelViewStateHandler.kt

```diff
@@ -0,0 +1,30 @@
+package sp.bvantur.tasky.core.presentation
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.withContext
+import sp.bvantur.tasky.core.DispatcherProvider
+
+interface ViewModelViewStateHandler<State : ViewState> {
+    val viewStateFlow: StateFlow<State>
+
+    suspend fun emitViewState(viewState: State)
+}
+
+class ViewModelViewStateHandlerImpl<State : ViewState>(
+    initialViewState: State,
+    private val dispatcherProvider: DispatcherProvider
+) : ViewModelViewStateHandler<State> {
+    private val mutableViewStateFlow = MutableStateFlow(initialViewState)
+    override val viewStateFlow: StateFlow<State> = mutableViewStateFlow.asStateFlow()
+
+    override suspend fun emitViewState(viewState: State) {
+        withContext(dispatcherProvider.main) {
+            mutableViewStateFlow.update {
+                viewState
+            }
+        }
+    }
```

Comment:
I have a similar opinion about this, but this is at least one of the better abstractions which actually has a single responsibility (and doesn't try to be a `BaseViewModel` for everything)

Reply:
If both of you don't see any misuse with this approach I would like to keep it. Even though some points are valid about having that prematurely and a need for some extra clicks to get to the actual code of what is doing it, I see also some benefits of having this.

1. I don't need to declare components for handling ViewState again and again, I can just use this abstraction for it. 
2. Actual implementation of how to update the state is taken care of inside of this component and I have 1 thing to worry about every time there is a need for updating a ViewState. It will always use `state.update{}` block and there is no way for me to update the state with some other mechanisms that are available with StateFlow.
3. Easy to plug and play this capability to any ViewModel wherever that is needed.
4. In my experience of using such a component in some of my projects, I intend to not need to investigate that code at all. It simply works and I don't need to think about it. If it is taken as a project component that has it purpose you just use it and that is it.
5. Easier maintenance or exchange of logic that is being used for ViewState handling all over the project.

Reply:
You don't have to stick to anything we say, we're just here encouraging some technical discussions and showing some other perspectives 😄 

The main issue I see with your approach is that you force state updates to happen in a suspending context. Not all state updates require that, yet with your function it has to be called in a coroutine.

Reply:
It's ok....I don't know why I asked that. It makes sense that a dispatcher provider can be used across the project.